src/EhsnPlugin/Mappers/FieldVisitMapper.cs

```diff
@@ -156,60 +157,60 @@ private string MapWeather()
         {
             if (_eHsn.EnvCond == null) return null;
 
-            var lines = new List<string>();
+            var stringBuilder = new StringBuilder();
 
-            AddCommentLine(lines, "Cloud Cover", _eHsn.EnvCond.cloudCover);
-            AddCommentLine(lines, "Precipitation", _eHsn.EnvCond.precipitation);
-            AddCommentLine(lines, "Wind Conditions", _eHsn.EnvCond.windMagnitude);
-            AddCommentLine(lines, "Wind Speed", _eHsn.EnvCond.windMagnitudeSpeed);
-            AddCommentLine(lines, "Wind Direction", _eHsn.EnvCond.windDirection);
+            AddCommentLine(stringBuilder, "Cloud Cover", _eHsn.EnvCond.cloudCover);
+            AddCommentLine(stringBuilder, "Precipitation", _eHsn.EnvCond.precipitation);
+            AddCommentLine(stringBuilder, "Wind Conditions", _eHsn.EnvCond.windMagnitude);
+            AddCommentLine(stringBuilder, "Wind Speed", _eHsn.EnvCond.windMagnitudeSpeed);
+            AddCommentLine(stringBuilder, "Wind Direction", _eHsn.EnvCond.windDirection);
 
-            return string.Join("\n", lines);
+            return stringBuilder.ToString().Trim();
         }
 
-        private static void AddCommentLine(List<string> lines, string label, string value)
+        private static void AddCommentLine(StringBuilder stringBuilder, string label, string value)
         {
             if (string.IsNullOrWhiteSpace(value))
                 return;
 
-            var valueLines = value.Trim().Split('\n');
+            var trimmedValue = value.Trim();
 
-            if (valueLines.Length == 1)
+            // Single-line values map to a single line in the comment
+            if (trimmedValue.Count(v => v == '\n') == 0)
             {
-                // Single-line values map to a single line in the comment
-                lines.Add($"{label}: {valueLines[0]}");
+                stringBuilder.AppendFormat("{0}: {1}\n", label, trimmedValue);
                 return;
             }
 
             // Multi-line values are de-marked
-            lines.Add($"== {label}:");
-            lines.AddRange(valueLines);
-            lines.Add("==");
+            stringBuilder.AppendFormat("== {0}:\n", label);
+            stringBuilder.AppendFormat("{0}\n", trimmedValue);
+            stringBuilder.Append("==\n");
```

Comment:
Same here:
```suggestion
            stringBuilder.AppendLine($"== {label}:");
            stringBuilder.AppendLine("${trimmedValue}");
            stringBuilder.AppendLine("==");
```
Or:
```suggestion
            stringBuilder.Append($"== {label}:\n");
            stringBuilder.Append("${trimmedValue}\n");
            stringBuilder.Append("==\n");
```

Reply:
same as above comment; i don't think it makes sense to use string interpolation with stringbuilder